File: leetcode/1512-number-of-good-pairs/solution.cpp
/**
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem: https://leetcode.com/problems/number-of-good-pairs/
 * Runtime: 0ms (100.00%)
 */

const static auto _ = []() {
    cin.tie(nullptr)->sync_with_stdio(false);
    return nullptr;
}();

class Solution {
  public:
    int numIdenticalPairs(vector<int> &nums) {
        int goodPairsNum = 0;
        int n = nums.size();

        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                if (nums[i] == nums[j]) {
                    goodPairsNum++;
                }
            }
        }
        return goodPairsNum;
    }
};

File: leetcode/1512-number-of-good-pairs/solution.go
/**
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem: https://leetcode.com/problems/number-of-good-pairs/
 * Runtime: 0ms (100.00%)
 */

func numIdenticalPairs(nums []int) int {
	goodPairsNum := 0

	for i := 0; i < len(nums); i++ {
		for j := i + 1; j < len(nums); j++ {
			if nums[j] == nums[i] {
				goodPairsNum++
			}
		}
	}

	return goodPairsNum
}
